Game/Assets/Shaders/depthMap.shader

```diff
@@ -0,0 +1,42 @@
+--- vertDepthMap
+
+#define MAX_BONES 100
+
+in layout(location=0) vec3 pos;
+in layout(location=1) vec3 norm;
+in layout(location=4) uvec4 boneIndices;
+in layout(location=5) vec4 boneWeitghts;
+
+uniform mat4 model;
+uniform mat4 view;
+uniform mat4 proj;
+
+uniform mat4 palette[MAX_BONES];
+uniform bool hasBones;
+
+out vec2 texcoords;
+
+void main() {
+
+	vec4 position = vec4(pos, 1.0);
+    vec4 normal = vec4(norm, 0.0);
+
+    if (hasBones)
+    { 
+        mat4 skinT = palette[boneIndices[0]] * boneWeitghts[0] + palette[boneIndices[1]] * boneWeitghts[1]
+        + palette[boneIndices[2]] * boneWeitghts[2] + palette[boneIndices[3]] * boneWeitghts[3];
+
+        position = skinT * vec4(pos, 1.0);
+        normal = skinT * vec4(norm, 0.0);
+    }
+
+	gl_Position = proj * view * model * position;
+}
+
+--- fragDepthMap
+
+void main() {
+	
+	// gl_FragDepth = gl_FragCoord.z;
+
```

Comment:
This shader has no fragment shader. Is this correct?

Reply:
Yes. For shadow mapping, you don't need to colour nothing.